Controllers/AdminController.cs

```diff
@@ -51,24 +63,48 @@ public IActionResult Index(long id)
         public IActionResult CreateBook()
         {
             //Form uses the collection of Writers to populate the drop-down list
-            return View(new CreateBookVM { Writers = _db.Writers });
+            return View(new CreateBookVM { Writers = _db.Writers.OrderBy(b => b.Name) });
         }
         [HttpPost]
         public IActionResult CreateBook(CreateBookVM bookVM, long id)
         {
             Writer writer;
+            if (bookVM.AuthorId > 0)
+            {
+                writer = _db.Writers.Single(b => b.Id == bookVM.AuthorId);
+            }
+            else
+            {
+                writer = new Writer { Name = bookVM.Name };
+                _db.Add(writer);
+                _db.SaveChanges();
+            }
             //TODO: Assign the writer object depending on what is entered for the AUTHOR NAME in the ViewModel
             // if the VM has an AuthorId (from the drop-down) you can use it to get that writer, otherwise create a new one
 
-
+            Book book = new Book { Title = bookVM.Title, Author = writer, SKU = bookVM.SKU, Price = bookVM.Price };
             //TODO: Create a new Book (with the writer above)
-
+            if(id > 0)
+            {
+                var bookToUpdate = new Book { Title = book.Title, Author = book.Author, SKU = book.SKU, Id = id, Price = book.Price };
+                _db.Update(bookToUpdate);
+                _db.SaveChanges();
+                return RedirectToAction("Index", new {id = bookToUpdate.Id});
+            }
+            else
+            {
+                _db.Add(book);
+                _db.SaveChanges();
+                return RedirectToAction("Index", new { id = book.Id });
+            }
```

Comment:
The logic here works great. 

Once you have it working, a good practice is to refactor code to make it simpler, for example, you could reuse the `book` object (you can set it's id using `book.Id = id`) and move the two repeated lines outside the condition, to the bottom: 
`_db.SaveChanges();` 
`return RedirectToAction("Index", new { id = book.Id});`